_posts/ref/2022-01-18-wire-graph-service.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: page
+title:  "WireGraphService Configuration Format"
+categories: [ref]
+---
+
+This document describes the format of the configuration of the WireGraphService component. The configuration of the WireGraphService contains the informations related to the Wire Graph topology and rendering properties.
+The pid of the WireGraphService configuration is `org.eclipse.kura.wire.graph.WireGraphService`.
+The WireGraphService configuration contains a single property of string type named `WireGraph` that contains a serialized JSON representation of a [WireGraph](#wire_graph) object describing the current graph layout.
+
+## JSON definitions
+#### Position
+An object representing a Wire Component position
+<br>**Properties**:
+
+* **x**: `number`
+  * **optional** If not specified, 0.0 will be used as default value
+    The x coordinate of the Wire Component inside the graph canvas
+* **y**: `number`
+  * **optional** If not specified, 0.0 will be used as default value
+    The y coordinate of the Wire Component inside the graph canvas
+
+```json
+{
+  "x": 40,
+  "y": 0
+}
+```
+```json
+{
+  "x": 1.5
+}
+```
+```json
+{}
+```
+#### PortNameList
+An object that specifies custom names for Wire Component input and output ports. The name of properties of this object must be an integer starting from 0, representing the index of the port whose name needs to be assigned. If the name of a specific property is not specified, the default port name will be used.
```

Comment:
> The name of properties of this object must be an integer starting from 0, representing the index of the port whose name needs to be assigned.

The properties name for this object must be represented as an integer starting from 0, matching the index of the port whose name needs to be assigned.